x-pack/legacy/plugins/reporting/test_helpers/create_mock_browserdriverfactory.ts

```diff
@@ -0,0 +1,129 @@
+/*
```

Comment:
Whew this is a fun one

Reply:
Yeah.. I think this test and the way things are mocked, especially in `createPage` highlight some overly-complicated areas of Reporting and maybe things that have real performance impact.